org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/debug/FqnConverter.java

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.jdt.ls.extension.core.internal.debug;
+
+import static java.lang.String.format;
+
+import com.google.common.base.Preconditions;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.commons.lang3.tuple.Pair;
+import org.eclipse.che.jdt.ls.extension.api.dto.LocationParameters;
+import org.eclipse.che.jdt.ls.extension.core.internal.JavaModelUtil;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.jdt.core.IClassFile;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IMember;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.search.IJavaSearchConstants;
+import org.eclipse.jdt.core.search.IJavaSearchScope;
+import org.eclipse.jdt.core.search.SearchEngine;
+import org.eclipse.jdt.core.search.TypeNameMatch;
+import org.eclipse.jdt.core.search.TypeNameMatchRequestor;
+import org.eclipse.jdt.internal.core.JavaProject;
+import org.eclipse.jdt.internal.core.SourceMethod;
+import org.eclipse.jdt.internal.core.SourceType;
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.Document;
+import org.eclipse.jface.text.IRegion;
+
+/** @author Anatolii Bazko */
+public class FqnConverter {
+
+  /** Converts {@link LocationParameters} into fqn. The first parameter contains */
+  public static String locationToFqn(List<Object> params, IProgressMonitor pm) {
+    Preconditions.checkArgument(params.size() >= 2, "File uri and line number are expected");
+
+    if (pm.isCanceled()) {
+      throw new OperationCanceledException();
+    }
+
+    final String fileUri = (String) params.get(0);
+    final Integer lineNumber = Integer.valueOf(params.get(1).toString());
+
+    // Not a URI then let's assume it is a FQN.
+    if (!fileUri.startsWith("file:")) {
+      return fileUri;
+    }
+
+    IJavaProject javaProject = JavaModelUtil.getJavaProject(fileUri);
+
+    if (javaProject == null) {
+      throw new IllegalArgumentException(format("Project for '%s' not found", fileUri));
+    }
+
+    URI fileRelativeUri = javaProject.getProject().getLocationURI().relativize(URI.create(fileUri));
```

Comment:
You can use JDTUtils#toURI to get URI
https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java#L653